app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/EventCard.kt

```diff
@@ -17,19 +17,25 @@ import de.tum.`in`.tumcampusapp.component.ui.overview.card.CardViewHolder
 import de.tum.`in`.tumcampusapp.component.ui.ticket.activity.EventDetailsActivity
 import de.tum.`in`.tumcampusapp.component.ui.ticket.adapter.EventsAdapter
 import de.tum.`in`.tumcampusapp.component.ui.ticket.model.Event
+import de.tum.`in`.tumcampusapp.component.ui.ticket.repository.EventsLocalRepository
+import de.tum.`in`.tumcampusapp.component.ui.ticket.repository.TicketsLocalRepository
 import de.tum.`in`.tumcampusapp.component.ui.tufilm.KinoActivity
+import de.tum.`in`.tumcampusapp.database.TcaDb
 import de.tum.`in`.tumcampusapp.utils.Const
 
 class EventCard(context: Context) : Card(CardManager.CARD_EVENT, context, "card_event") {
 
     var event: Event? = null
-    private val eventsController = EventsController(context)
+
+    // TODO(thellmund) Inject this
+    private val eventCardsProvider = EventCardsProvider(context, EventsLocalRepository(TcaDb.getInstance(context)))
```

Comment:
WIP?

Reply:
Might be a little more work. Thought about addressing injections in cards in a separate PR.